interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -142,21 +157,64 @@ object TapirAdapter {
             enableIntrospection = enableIntrospection,
             queryExecution
           )
-      ).either
+      ).map(buildResponse).either
     }
 
     makeHttpEndpoints.map(_.serverLogic(logic))
   }
 
+  private[caliban] def buildResponse[E](response: GraphQLResponse[E])(implicit
+    responseCodec: JsonCodec[ResponseValue]
+  ) =
+    response match {
+      case resp @ GraphQLResponse(StreamValue(stream), _, _, _) =>
+        val Newline     = "\r\n"
+        val ContentType = "Content-Type: application/json; charset=utf-8"
+        val SubHeader   = s"$Newline$ContentType$Newline$Newline"
+        val Boundary    = "---"
+        val theBoundary = s"$Newline$Boundary$SubHeader"
+        val endBoundary = s"$Newline-----$Newline"
+
+        val encoded = ZStream
+          .unwrapScoped(
+            for {
+              initialAndSubsequent <- stream.peel(ZSink.head[ResponseValue])
+              (initial, subsequent) = initialAndSubsequent
+              payload               = ZStream.fromIterable(
+                                        initial.map(value => resp.copy(data = value).toResponseValue)
+                                      ) ++ subsequent
+            } yield payload
+              .map(responseCodec.encode)
+              .intersperse(theBoundary, theBoundary, endBoundary)
+          )
+          .mapConcat(_.getBytes(StandardCharsets.UTF_8))
+
+        (
+          MediaType.MultipartMixed.copy(otherParameters = Map("boundary" -> "-", "deferSpec" -> "20220824")),
+          encoded
+        )
+
+      case response =>
+        val encodedResponse =
+          ZStream.fromIterable(responseCodec.encode(response.toResponseValue).getBytes(StandardCharsets.UTF_8))
```

Comment:
I was just scrolling through this PR since it's quite interesting (great job by the way!) and this happened to catch my gaze.

For my own understanding, wouldn't this cause the response to be streamed in chunks of 1 byte? If yes, would it be better to use `ZStream.succeed` instead or there shouldn't be any overhead doing this?

Reply:
`[ZStream.fromIterable](https://github.com/zio/zio/blob/d5df88d8860d544e138c4f06972ab2b6314083bc/streams/shared/src/main/scala/zio/stream/ZStream.scala#L4185-L4216)` iterates the given `Array[Byte]` and splits it into chunks of the default size (4096 currently.)

Although it's a reasonable size, I'd suggest wrapping the array directly using `Chunk.fromArray` to avoid copy.

`ZStream.fromChunk(Chunk.fromArray(responseCodec.encode(response.toResponseValue).getBytes(StandardCharsets.UTF_8)))`

Reply:
@guersam fixed